### PR TITLE
Introduce variable BASE_VERSION

### PR DESCRIPTION
**Pull request by Soulofdestiny**

### CODE REVIEW

**Comment by Soulofdestiny** [comment]
@okurz @coolo 
as discussed some time ago in https://progress.opensuse.org/issues/12330#note-18

I would like to have feedback

File: products/sle/main.pm [hunk 1]
```diff
@@ -57,19 +57,19 @@ sub is_update_test_repo_test {
 
 sub cleanup_needles {
     remove_common_needles;
-    if (get_var('VERSION', '') ne '12') {
+    if ((get_var('VERSION', '') ne '12') && (!get_var('BASE_VERSION', '') eq '12')) {
```
Comment (okurz) [inline products/sle/main.pm]: can you use the same syntax for both comparisons, i.e.

```perl
(get_var('VERSION', '') ne '12') && (get_var('BASE_VERSION', '') ne '12')
```
or is my logic wrong here?

**Comment by Soulofdestiny** [comment]
updated

**Comment by okurz** [comment]
+1

**Comment by QingmingSu** [comment]
This week I have got some failures during sle12sp1/sp2+addon installations due to needle issue: sp1/sp2 needles are unregistered because the jobs that installation of sle12sp1/sp2 as parent jobs of upgrade tests on s390x are triggered by sp3 ISO which means VERSION=12-SP3.

May be one workaround is creating version independent needles, but so far I have already observed several needles need fix, like release-notes-ha needle for sle12sp1, release-notes-sdk needle for sle12sp2, addon-license-ha for sle12sp2, ..... 

Seems the version independent needle isn't good option compared to BASE_VERSION.

**Comment by Soulofdestiny** [comment]
@coolo @sysrich @aaannz

**Comment by coolo** [comment]
this is definitely better than unversioned needles, so go ahead with the plan

**Pull request merged**